eng/build.sh

```diff
@@ -402,6 +403,16 @@ while [[ $# > 0 ]]; do
       shift 1
       ;;
 
+     -distrorid)
+      if [ -z ${2+x} ]; then
+        echo "No value for distrorid is supplied. See help (--help) for supported values." 1>&2
+        exit 1
+      fi
+      __DistroRid="$(echo "$2" | tr "[:upper:]" "[:lower:]")"
+      arguments="$arguments /p:PackageRid=$__DistroRid"
```

Comment:
@am11 if I understood correctly, I should be using `OutputRid` here.
But that's not working well:
```
$ ./build.sh  --distrorid banana-x64
__DistroRid: banana-x64
  Determining projects to restore...
  Tool 'coverlet.console' (version '3.1.2') was restored. Available commands: coverlet
  Tool 'dotnet-reportgenerator-globaltool' (version '5.0.2') was restored. Available commands: reportgenerator
  Tool 'microsoft.dotnet.xharness.cli' (version '1.0.0-prerelease.22411.1') was restored. Available commands: xharness
  Tool 'microsoft.visualstudio.slngen.tool' (version '8.1.6') was restored. Available commands: slngen
  Tool 'dotnet-format' (version '6.0.240501') was restored. Available commands: dotnet-format
  
  Restore was successful.
  All projects are up-to-date for restore.
  Determining projects to restore...
/home/tmds/repos/runtime/.dotnet/sdk/7.0.100-preview.7.22377.5/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(90,5): error NETSDK1083: The specified RuntimeIdentifier 'banana-x64' is not recognized. [/home/tmds/repos/runtime/Build.proj]

Build FAILED.

/home/tmds/repos/runtime/.dotnet/sdk/7.0.100-preview.7.22377.5/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(90,5): error NETSDK1083: The specified RuntimeIdentifier 'banana-x64' is not recognized. [/home/tmds/repos/runtime/Build.proj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:11.57
Build failed with exit code 1. Check errors above.
```
The rid does get used to restore something.

Reply:
I changed ILCompiler.csproj to use PackageRid and now I get past this restore.
I've pushed the change to see what CI makes of it.